src/hip_hcc.cpp

```diff
@@ -1935,9 +1935,9 @@ void ihipStream_t::resolveHcMemcpyDirection(unsigned hipMemKind,
             throw ihipException(hipErrorRuntimeOther);
     };
 
-    if (srcPtrInfo->_isInDeviceMem) {
+    if (srcPtrInfo->_isInDeviceMem && srcPtrInfo->_appId != -1) {
```

Comment:
Is there a case where !_isInDeviceMem and appId != -1 ?  Wonder if the second comparison is redundant.

Reply:
@bensander It is not possible to have _appId = -1 and _isInDeviceMem = true.

Reply:
@bensander @gargrahul I will remove the second check and update the PR. Thanks.